lib/src/main/java/com/dynatrace/file/util/PollBasedFilePoller.java

```diff
@@ -54,30 +56,24 @@ public Thread newThread(Runnable r) {
   @Override
   public boolean fileContentsUpdated() {
     // get the current value and reset to false
-    return changedSinceLastPoll.getAndSet(false);
+    return changedSinceLastInquiry.getAndSet(false);
   }
 
-  private void poll() {
-    synchronized (this) {
-      try {
-        long modifiedAt = Files.getLastModifiedTime(absoluteFilePath).toMillis();
-        long previouslyModifiedAt = lastUpdatedAt;
-        lastUpdatedAt = modifiedAt;
+  private synchronized void poll() {
+    try {
+      FileTime lastModifiedAt = Files.getLastModifiedTime(absoluteFilePath);
 
-        if (modifiedAt >= previouslyModifiedAt) {
-          if (previouslyModifiedAt > 0) {
-            changedSinceLastPoll.set(true);
-          }
+      if (lastModifiedAt.compareTo(prevModifiedAt) >= 0) {
+        if (prevModifiedAt.compareTo(ZERO_FILE_TIME) != 0) {
```

Comment:
`!equals()` would do the same but is more explicit/readable. It could also be combined in one `if`.